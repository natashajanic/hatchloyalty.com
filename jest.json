{
  "collectCoverage": true,
  "coveragePathIgnorePatterns": ["tests"],
  "globals": {},
  "moduleFileExtensions": [
    "ts",
    "tsx",
    "js"
  ],
  "moduleNameMapper": {
    "^.+\\.(css|less|scss)$": "identity-obj-proxy",
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/tests/assetsTransformer.ts",
    "^src/(.*)$": "<rootDir>/src/$1"
  },
  "setupFiles": [
    "raf/polyfill"
  ],
  "setupFilesAfterEnv": [
    "<rootDir>/tests/setupTests.ts"
  ],
  "snapshotSerializers": [
    "enzyme-to-json/serializer"
  ],
  "transform": {
    "\\.(ts|tsx)$": "ts-jest"
  },
  "testRegex": "/src/.*/__tests__/.*\\.(ts|tsx|js)$",
  "unmockedModulePathPatterns": [],
  "verbose": true
}
