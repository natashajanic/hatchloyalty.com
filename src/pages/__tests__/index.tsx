import { shallow, ShallowWrapper } from 'enzyme'
import { Link } from 'gatsby'
import 'jest' // tslint:disable-line no-import-side-effect
import * as React from 'react'
import Wrapper from 'src/components/wrapper'
import { PureIndexPage, IPureIndexPageProps } from '../index'

describe('IndexPage', () => {
  let props: IPureIndexPageProps
  let mountedComponent: ShallowWrapper<any, any, any>

  beforeEach(() => {
    props = {
      data: {
        allMarkdownRemark: {
          edges: [
            {
              node: {
                excerpt: 'Test post 1 excerpt.',
                fields: { slug: '/test-post-1/' },
                frontmatter: { title: 'Test Post 1' },
              },
            },
            {
              node: {
                excerpt: 'Test post 2 excerpt.',
                fields: { slug: '/test-post-2/' },
                frontmatter: { title: 'Test Post 2' },
              },
            },
            {
              node: {
                excerpt: 'Test post 3 excerpt.',
                fields: { slug: '/test-post-3/' },
                frontmatter: { title: 'Test Post 3' },
              },
            },
          ],
        },
      },
    }
    mountedComponent = shallow(<PureIndexPage {...props} />)
  })

  describe('render', () => {
    it('displays a page title', async () => {
      expect(mountedComponent.find('h2').length).toEqual(1)
      expect(mountedComponent.find('h2').text()).toContain('Hello, Hatch')
    })

    it('displays links to the features section', async () => {
      expect(mountedComponent.find(Link).length).toEqual(3)
      expect(
        mountedComponent
          .find(Link)
          .at(0)
          .prop('to')
      ).toEqual('/features')
      expect(
        mountedComponent
          .find(Link)
          .at(1)
          .prop('to')
      ).toEqual('/features')
      expect(
        mountedComponent
          .find(Link)
          .at(2)
          .prop('to')
      ).toEqual('/features')
    })

    it('renders a teaser for each of the features posts', async () => {
      expect(mountedComponent.find(Wrapper).text()).toContain('Test Post 1')
      expect(mountedComponent.find(Wrapper).text()).toContain(
        'Test post 1 excerpt.'
      )
      expect(mountedComponent.find(Wrapper).text()).toContain('Test Post 2')
      expect(mountedComponent.find(Wrapper).text()).toContain(
        'Test post 2 excerpt.'
      )
      expect(mountedComponent.find(Wrapper).text()).toContain('Test Post 3')
      expect(mountedComponent.find(Wrapper).text()).toContain(
        'Test post 3 excerpt.'
      )
    })

    it('matches the expected snapshot', async () => {
      expect(mountedComponent).toMatchSnapshot()
    })
  })
})
