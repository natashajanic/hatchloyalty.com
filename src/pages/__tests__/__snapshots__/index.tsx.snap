// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage render matches the expected snapshot 1`] = `
<Layout>
  <SEO
    keywords={
      Array [
        "hatch",
        "hatch loyalty",
        "loyalty",
        "personalization",
        "activation",
        "data",
        "platform",
        "security",
        "reliable",
      ]
    }
    title="Home"
  />
  <Styled(Component)
    blacklist={
      Array [
        "css",
        "maxWidth",
        "m",
        "mt",
        "mr",
        "mb",
        "ml",
        "mx",
        "my",
        "p",
        "pt",
        "pr",
        "pb",
        "pl",
        "px",
        "py",
      ]
    }
    is="article"
    m="0 auto"
    maxWidth="1024px"
    px={
      Array [
        3,
        4,
      ]
    }
    py={5}
  >
    <h2>
      Goodbye, traditional loyalty. Hello, Hatch.
    </h2>
    <h3>
      Technology designed to help you build stronger, more personal relationships with your customers.
    </h3>
    <ForwardRef
      to="/features"
    >
      Learn More
    </ForwardRef>
    <section>
      <h3>
        Our Approach
      </h3>
      <h4>
        A modern platform for the modern organization.
      </h4>
      <div>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Build
          </h4>
          <p>
            Designing the perfect loyalty program is overwhelming. With Hatch, you don’t have to figure out all of the bells & whistles today. Start small and iterate toward the best program for your business.
          </p>
        </Styled(Component)>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Measure
          </h4>
          <p>
            Good loyalty programs leverage member data to make more informed decisions. With Hatch, the best programs can understand engagement and make those decisions in real-time.
          </p>
        </Styled(Component)>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Learn
          </h4>
          <p>
            Build a real-time feedback loop with your members. With Hatch, access actionable insights that can be used to drive engagement and change customer behavior
          </p>
        </Styled(Component)>
        <ForwardRef
          to="/features"
        >
          How it Works
        </ForwardRef>
      </div>
    </section>
    <section>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
      >
        <h3>
          Why we're differnt
        </h3>
        <h4>
          Open, Agile, Extensible
        </h4>
        <p>
          Traditional loyalty programs are overly complicated, inflexible, and expensive. Hatch's platform approach to loyalty is designed to integrate with existing systems, products, and partners. Let us help you get to market faster and cheaper while still providing the flexibility to adapt into the future.
        </p>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
      >
        <img
          alt=""
        />
      </Styled(Component)>
    </section>
    <section>
      <div>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Flexible
          </h4>
          <p>
            Hatch provides a simple API and a robust set of tools, giving you the control necessary to build the program your customers deserve.
          </p>
        </Styled(Component)>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Scalable
          </h4>
          <p>
            Hatch is built to grow with your business. Whether it's one location or thousands, Hatch dynamically scales with your traffic.
          </p>
        </Styled(Component)>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Reliable
          </h4>
          <p>
            We take pride in our platform, which is extensively tested and monitored to ensure your program is always up and running.
          </p>
        </Styled(Component)>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h4>
            Portable
          </h4>
          <p>
            Access to your data when and where you need it most. Hatch securely collects and shares data with your business in real-time.
          </p>
        </Styled(Component)>
      </div>
      <div>
        <Styled(Component)
          blacklist={
            Array [
              "css",
              "alignItems",
              "border",
              "borderColor",
              "borderBottom",
              "borderRadius",
              "color",
              "bg",
              "display",
              "justifyContent",
              "m",
              "mt",
              "mr",
              "mb",
              "ml",
              "mx",
              "my",
              "p",
              "pt",
              "pr",
              "pb",
              "pl",
              "px",
              "py",
              "width",
            ]
          }
        >
          <h3>
            Developer Friendly
          </h3>
          <h4>
            Technology you can trust
          </h4>
          <p>
            Hatch’s cloud-based platform provides you with best in class security and an unmatched ability to scale. We care about the developer experience. Leave your worries in our hands and focus on your bottom line -- we’ve got you covered.
          </p>
          <ul>
            <li>
              Extensive Documentation
            </li>
            <li>
              Data Extracts
            </li>
            <li>
              Webhooks
            </li>
            <li>
              Test Environments
            </li>
            <li>
              Isolated Runtimes
            </li>
            <li>
              Enterprise-Grade Support
            </li>
          </ul>
          <a
            href="https://developer.hatchloyalty.com"
          >
            Explore Documentation
          </a>
        </Styled(Component)>
      </div>
    </section>
    <section>
      <h3>
        Designed for Integration
      </h3>
      <h4>
        Use the best tool for the job
      </h4>
      <p>
        Avoid legacy lock-in. Hatch is designed to easily integrate with existing business systems and complimentary products & services, giving you the flexibility and control to build the best solution for your business.
      </p>
      <ForwardRef
        to="/features"
      >
        Learn More
      </ForwardRef>
    </section>
    <section>
      <h3>
        Our Blog
      </h3>
      <h4>
        Learn How Hatch Is Changing the Way Companies Approach Loyalty and Customer Engagment
      </h4>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        key="/test-post-1/"
      >
        <h3>
          Test Post 1
        </h3>
        <p>
          Test post 1 excerpt.
        </p>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        key="/test-post-2/"
      >
        <h3>
          Test Post 2
        </h3>
        <p>
          Test post 2 excerpt.
        </p>
      </Styled(Component)>
      <Styled(Component)
        blacklist={
          Array [
            "css",
            "alignItems",
            "border",
            "borderColor",
            "borderBottom",
            "borderRadius",
            "color",
            "bg",
            "display",
            "justifyContent",
            "m",
            "mt",
            "mr",
            "mb",
            "ml",
            "mx",
            "my",
            "p",
            "pt",
            "pr",
            "pb",
            "pl",
            "px",
            "py",
            "width",
          ]
        }
        key="/test-post-3/"
      >
        <h3>
          Test Post 3
        </h3>
        <p>
          Test post 3 excerpt.
        </p>
      </Styled(Component)>
    </section>
    <section>
      <blockquote>
        “Hatch is a robust, but easy to integrate customer loyalty platform that enables retailers to better understand and incentivize their customers…allowing them to engage their customers and grow their sales in a meaningful way.”
        <cite>
          Jeff Jordan General Partner, Andreessen Horowitz
        </cite>
      </blockquote>
    </section>
    <section>
      <h3>
        Try the Hatch Platform!
      </h3>
      <h4>
        Contact us to learn more.
      </h4>
      <ActiveCampaignForm
        formID="_form_1"
        formSrc="https://hatchloyalty.activehosted.com/f/embed.php?id=1"
      />
    </section>
  </Styled(Component)>
</Layout>
`;
